### PR TITLE
Library

### PR DESCRIPTION
**Pull request by Rfpnj**
Reworks the Library to be a bit more organized, Races are split into their home areas, each area of shelves has its own flavor that reflects the shelves. Allows for future slave interaction. 

I have removed the old post reading options and elected to return to the top-level of the library to reflect that you would need to walk back into the stacks.

### CODE REVIEW

**Comment by Rfpnj** [comment]
I went and cleaned up the branch a bit to reduce number of effected files

**Comment by Pimvgd** [comment]
If you rebase this on master and then remove the two commits that cancel each other out you're left with one commit, see https://github.com/Pimvgd/liliths-throne-public/commits/library as an example

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 1]
```diff
@@ -28,16 +28,100 @@ public String getContent() {
...
 		@Override
-		public Response getResponse(int index) {
-			if (index == 1) {
+		public Response getResponse(int main) {
+			if (main == 1) {
+				return new Response("Browse the Stacks", "Read one of the many books availble in the library.", BROWSE_BOOKS) {
+				};
+
+			} else if (main == 2) {
+				return new Response("Interact with Slaves", "Not yet implimented", null) {
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: implimented -> implemented

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 2]
```diff
@@ -28,16 +28,100 @@ public String getContent() {
...
+
+		@Override
+		public String getLabel() {
+			return "Library";
+		}
+
+		@Override
+		public String getContent() {
+			return "<p>"
+					+ "Walking down one of the aisles, you see a great deal of organization has gone into the design of the room. It clear that someone has spent a great deal of cleaning the room."
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: "It clear that" -> "It's clear that"/"It is clear that"

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 3]
```diff
@@ -28,16 +28,100 @@ public String getContent() {
...
+					+ " As you walk you scan the titles printed onto the spines of the books, but there's not really much that catches your eye."
+					+ " Only a few shelves really look to be worth your time, and you wonder if you should take some time to do a spot of reading..."
+					+ "</p>";
+							
+		}
+
+		@Override
+		public Response getResponse(int books) {
+			if (books == 1) {
+				return new Response("General Knowledge", "A section of the library dedicated to books on common subjects", LORE_BOOKS);
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: missing period at end (also for all the other books)

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 4]
```diff
@@ -28,16 +28,100 @@ public String getContent() {
...
+		public String getLabel() {
+			return "Library";
+		}
+
+		@Override
+		public String getContent() {
+			return "<p>"
+					+ "Walking down one of the aisles, you see that these shelves have an arcane look to them. The shelves glow and shift before your eyes."
+					+ "The enhanced sight you seem to have gained allows you to see layers of arcane essance has been built up on these shelves. Some of the texts almost look like"
+					+ "they are about to jump off the shelves under their own power."
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: missing space, will resolve to "... almost look likethey are about to..."

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 5]
```diff
@@ -28,16 +28,100 @@ public String getContent() {
...
+		@Override
+		public String getLabel() {
+			return "Library";
+		}
+
+		@Override
+		public String getContent() {
+			return "<p>"
+					+ "Walking down one of the aisles, you see that these shelves have an arcane look to them. The shelves glow and shift before your eyes."
+					+ "The enhanced sight you seem to have gained allows you to see layers of arcane essance has been built up on these shelves. Some of the texts almost look like"
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: "essance" -> "essence"

Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: "The enhanced sight you seem to have gained allows you to see layers of arcane essance has been built up"

something about this seems off... but I'm not sure. Is there a "that" missing somewhere or is it just the many different tenses making this look weird

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 6]
```diff
@@ -69,49 +153,15 @@ public void effects() {
...
-			} else if (index == 9) {
-				return bookResponse(ItemType.BOOK_HUMAN, Race.HUMAN);
-
-			} else if (index == 10) {
-				return bookResponse(ItemType.BOOK_WOLF_MORPH, Race.WOLF_MORPH);
-
-			} else if (index == 11) {
-				return bookResponse(ItemType.BOOK_SQUIRREL_MORPH, Race.SQUIRREL_MORPH);
+			} else if (lore == 0) {
+				return new Response("Back to the shelves", "Return to strolling the stacks", BROWSE_BOOKS);
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: missing period at end

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 7]
```diff
@@ -149,14 +199,10 @@ public String getContent() {
...
 		@Override
-		public Response getResponse(int index) {
-			if (index == 1) {
-				return new Response("Arcane Arousal", "A leather-bound tome that seems to offer an insight into how the arcane works.", null);
-
-			} else {
-				return LIBRARY.getResponse(index);
-			}
+		public Response getResponse(int lore) {
+			return  LIBRARY.getResponse(lore);
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: ... there's a double space here between "return" and "LIBRARY"

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 8]
```diff
@@ -198,13 +244,8 @@ public String getContent() {
...
 		@Override
-		public Response getResponse(int index) {
-			if (index == 2) {
-				return new Response("Lilith's Dynasty", "A hardback book that might give some clues as to who exactly Lilith is.", null);
-
-			} else {
-				return LIBRARY.getResponse(index);
-			}
+		public Response getResponse(int lore) {
+			return  LIBRARY.getResponse(lore);
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: same here (double space)

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 9]
```diff
@@ -243,13 +284,220 @@ public String getContent() {
 		}
 
 		@Override
-		public Response getResponse(int index) {
-			if (index == 3) {
-				return new Response("Dominion's History", "A paperback book describing the events that led to the creation of the city you currently find yourself in.", null);
+		public Response getResponse(int lore) {
+			return  LIBRARY.getResponse(lore);
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: same here (double space)

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 10]
```diff
@@ -243,13 +284,220 @@ public String getContent() {
...
+		@Override
+		public String getLabel() {
+			return "Library";
+		}
+
+		@Override
+		public String getContent() {
+			return "<p>"
+					+ "Walking down one of the aisles, you see that these shelves have an urban look to them. The shelves look like they haven been crafted from concrete and iron."
+					+ "These books have the feeling of being about the city and it's people"
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: abrupt ending - missing period or missing content?

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 11]
```diff
@@ -243,13 +284,220 @@ public String getContent() {
...
+				return bookResponse(ItemType.BOOK_HORSE_MORPH, Race.HORSE_MORPH);
+
+			} else if (city == 6) {
+				return bookResponse(ItemType.BOOK_HUMAN, Race.HUMAN);
+
+			} else if (city == 7) {
+				return bookResponse(ItemType.BOOK_WOLF_MORPH, Race.WOLF_MORPH);
+
+			} else if (city == 0) {
+				return new Response("Back to the shelves", "Return to strolling the stacks", BROWSE_BOOKS);
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: missing period at end

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 12]
```diff
@@ -243,13 +284,220 @@ public String getContent() {
...
+			return "<p>"
+					+ "Walking down one of the aisles, you see that these shelves have an urban look to them. The shelves look like they haven been crafted from concrete and iron."
+					+ "These books have the feeling of being about the city and it's people"
+					+ "</p>";
+							
+		}
+
+		@Override
+		public Response getResponse(int city) {
+			if  (city == 1) {
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: double space between if and (city == 1)

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 13]
```diff
@@ -243,13 +284,220 @@ public String getContent() {
...
+		@Override
+		public String getLabel() {
+			return "Library";
+		}
+
+		@Override
+		public String getContent() {
+			return "<p>"
+					+ "Walking down one of the aisles, you see that these shelves have an pastoral look to them. The shelves look like they haven been crafted from blocks of sod supported by pieces of wood."
+					+ " These books have the feeling of being about the fields and it's people"
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: I guess it's a missing period then if this one is similar in style

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 14]
```diff
@@ -243,13 +284,220 @@ public String getContent() {
...
+		@Override
+		public Response getResponse(int field) {
+			if  (field == 1) {
+				return bookResponse(ItemType.BOOK_SQUIRREL_MORPH, Race.SQUIRREL_MORPH);
+
+//			} else if (field == 2) {
+//				return bookResponse(ItemType.BOOK_COW_MORPH, Race.COW_MORPH);
+
+			}  else if (field == 0) {
+				return new Response("Back to the shelves", "Return to strolling the stacks", BROWSE_BOOKS);
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: missing period

File: src/com/base/game/dialogue/places/dominion/lilayashome/Library.java [hunk 15]
```diff
@@ -243,13 +284,220 @@ public String getContent() {
...
+			return "<p>"
+					+ "Walking down one of the aisles, you see that these shelves have an pastoral look to them. The shelves look like they haven been crafted from blocks of sod supported by pieces of wood."
+					+ " These books have the feeling of being about the fields and it's people"
+					+ "</p>";
+							
+		}
+
+		@Override
+		public Response getResponse(int field) {
+			if  (field == 1) {
```
Comment (Pimvgd) [inline src/com/base/game/dialogue/places/dominion/lilayashome/Library.java]: double space

**Comment by Rfpnj** [comment]
Ran back through this and corrected my sick-addled coding.

**Comment by Pimvgd** [comment]
you missed one?

**Pull request merged**